internal/premis/premis.go

```
@@ -249,6 +249,25 @@ func addEventElement(PREMISEl *etree.Element, event Event) {
 		outcomeDetailNoteEl.CreateText(event.Summary.OutcomeDetail)
 	}
 
+	// The order of PREMIS entities is important (i.e. first objects, then
+	// events, then agents) so we need to find the correct place to insert this
+	// event element.
+	var index int
+	if events := PREMISEl.FindElements("./premis:event"); len(events) > 0 {
+		// If there are other events, add this event after the last one.
+		last := events[len(events)-1]
+		index = last.Index() + 1
+	} else if objs := PREMISEl.FindElements("./premis:object"); len(objs) > 0 {
+		// If there are no other events, add this event after the last object.
+		last := objs[len(objs)-1]
+		index = last.Index() + 1
+	} else {
+		// If no PREMIS objects exist, something has gone horribly wrong.
+		return
```

Should we return an error?